nix/default.nix

```
@@ -1,45 +1,44 @@
 { system ? builtins.currentSystem }:
 let
-  pkgs = import ./nixpkgs.nix {};
+  pkgs = (import ./nixpkgs.nix {
+    config = {
+      packageOverrides = pkg: {
+        gpgme = (static pkg.gpgme);
+        libassuan = (static pkg.libassuan);
+        libgpgerror = (static pkg.libgpgerror);
+        libseccomp = (static pkg.libseccomp);
+      };
+    };
+  });
 
-  static = pkg: pkg.overrideAttrs(old: {
-    configureFlags = (old.configureFlags or []) ++
+  static = pkg: pkg.overrideAttrs(x: {
+    configureFlags = (x.configureFlags or []) ++
       [ "--without-shared" "--disable-shared" ];
     dontDisableStatic = true;
     enableSharedExecutables = false;
     enableStatic = true;
   });
 
-  patchLvm2 = pkg: pkg.overrideAttrs(old: {
-    configureFlags = [
-      "--disable-cmdlib" "--disable-readline" "--disable-udev_rules"
-      "--disable-udev_sync" "--enable-pkgconfig" "--enable-static_link"
-    ];
-    preConfigure = old.preConfigure + ''
-      substituteInPlace libdm/Makefile.in --replace \
-        SUBDIRS=dm-tools SUBDIRS=
-      substituteInPlace tools/Makefile.in --replace \
-        "TARGETS += lvm.static" ""
-      substituteInPlace tools/Makefile.in --replace \
-        "INSTALL_LVM_TARGETS += install_tools_static" ""
-    '';
-    postInstall = "";
-  });
-
-  self = {
-    cri-o-static = (pkgs.cri-o.overrideAttrs(old: {
+  self = with pkgs; {
+    cri-o-static = (cri-o.overrideAttrs(x: {
       name = "cri-o-static";
       src = ./..;
-      BUILDTAGS="${old.BUILDTAGS} netgo";
-      EXTRA_LDFLAGS = ''-linkmode external -extldflags "-static -lm"'';
-      dontStrip = true;
-      # DEBUG = 1; # Uncomment this line to enable debug symbols in the binary
-    })).override {
-      gpgme = (static pkgs.gpgme);
-      libassuan = (static pkgs.libassuan);
-      libgpgerror = (static pkgs.libgpgerror);
-      libseccomp = (static pkgs.libseccomp);
-      lvm2 = (patchLvm2 (static pkgs.lvm2));
-    };
+      doCheck = false;
+      buildInputs = [
+        glibc
+        glibc.static
+        gpgme
+        libapparmor
+        libassuan
+        libgpgerror
+        libseccomp
+        libselinux
+      ];
+      prePatch = ''
+        export LDFLAGS='-static-libgcc -static'
+        export EXTRA_LDFLAGS='-linkmode external -extldflags "-static -lm"'
+        export BUILDTAGS='static netgo apparmor selinux seccomp exclude_graphdriver_btrfs exclude_graphdriver_devicemapper'
```

Let's do not remove the btrfs support.
```suggestion
        export BUILDTAGS='static netgo apparmor selinux seccomp exclude_graphdriver_devicemapper'
```

---

I remove btrfs (together with remove lvm2) because I found that:
1. No btrfs support RHEL/CentOS 8 (https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/considerations_in_adopting_rhel_8/file-systems-and-storage_considerations-in-adopting-rhel-8#btrfs-has-been-removed_file-systems-and-storage)
2. Even remove btrfs and lvm2, openSUSE 15.1 vagrant + kvm could running correctly

As long as podman seems didn't compile these 2 recently for repo package (please correct me), shall we keep the static binary simple and tiny for most common use cases?

---

Hm, I'm concerned we remove a feature others may already rely on. So I suggest in keeping the btrfs support for the static binary.

---

Hmm.... For people who really care about stability and support generally will go for repo packages... Not statically linked binary... We shall not forced ourselves to support ALL features by statically linked binary but cover +80% daily use cases (due to 80/20 principle)